cpp/src/arrow/buffer_builder.h

```diff
@@ -87,13 +87,8 @@ class ARROW_EXPORT BufferBuilder {
 
   /// \brief Return a capacity expanded by an unspecified growth factor
   static int64_t GrowByFactor(int64_t current_capacity, int64_t new_capacity) {
-    // NOTE: Doubling isn't a great overallocation practice
-    // see https://github.com/facebook/folly/blob/master/folly/docs/FBVector.md
-    // for discussion.
-    // Grow exactly if a large upsize (the caller might know the exact final size).
-    // Otherwise overallocate by 1.5 to keep a linear amortized cost.
-    // TODO: revisit this?  See comment in BufferOutputStream::Reserve.
-    return std::max(new_capacity, current_capacity * 3 / 2);
+    // Doubling capacity except for large Reserve requests
```

Comment:
consider commenting that doubling works well with JEMalloc?

Reply:
It seems to work well with the system allocator as well. I guess the 1.5 factor is a good choice for smallish allocations, but less so once the allocator (re)allocates entire pages.

Reply:
Also, why "except for large Reserve requests"?